### PR TITLE
Finalize removal of typed pointers

### PR DESCRIPTION
**Pull request by dwightguth**
Previously we still had some references to typed pointers in .ll files and in API calls that still created typed pointers when working with older versions of LLVM. This led to some invalid bitcode being generated in ways that wasn't necessarily a problem, but which caused problems when I was making changes to that code. This PR is a preliminary to further changes which removes all remaining references to the API constructor for typed pointers and removes all typed pointers from flat bitcode files.

### CODE REVIEW

**Review by Robertorosmaninho — CHANGES_REQUESTED** [review]
I made some comments to improve the code quality. Sorry for being pedantic.
Also, `runtime/main/main.ll` still contains typed pointers, can this be a problem in the future?

File: lib/codegen/EmitConfigParser.cpp [hunk 1]
```diff
@@ -574,8 +574,7 @@ static void emit_get_token(kore_definition *definition, llvm::Module *module) {
           = allocate_term(float_ty, case_block, "kore_alloc_floating");
       llvm::Function *init_float = get_or_insert_function(
           module, "init_float", llvm::Type::getVoidTy(ctx),
-          llvm::PointerType::getUnqual(float_ty),
-          llvm::PointerType::getUnqual(ctx));
+          llvm::PointerType::getUnqual(ctx), llvm::PointerType::getUnqual(ctx));
```
[the remark this answers is not in the document]
Comment (dwightguth) [reply lib/codegen/EmitConfigParser.cpp:577]: Yes; I made the changes completely procedurally, so cleanups like this didn't get done, but I will go ahead and handle it in places that I notice it being called in more than one place in a function.

[the remark this answers is not in the document]
Comment (Robertorosmaninho) [reply lib/codegen/EmitConfigParser.cpp:577]: Thank you! And sorry again for being pedantic!

**Comment by dwightguth** [comment]
Don't worry at all; your suggestions do all make the code much nicer; I don't mind doing it and appreciate the thorough review. I believe I have now addressed all the issues. I will mark you for re-review as soon as I confirm that the tests are still passing.

**Review by theo25 — APPROVED** [review]
Looks good to me

**Review by Robertorosmaninho — APPROVED** [review]
Thank you for addressing all comments! The code looks much nicer indeed!
LGTM!

**Pull request merged**